p2p/channel.go

```diff
@@ -511,6 +519,24 @@ func (c *channel) setUpnpPortsRelease(release []func()) {
 	c.upnpPortsRelease = release
 }
 
+func (c *channel) checkIfChannelAlive() {
+	for {
+		select {
+		case <-c.stop:
+			return
+		case <-time.After(initialTrafficTimeout):
+			if c.remoteAlive {
```

Comment:
you're gonna race here. Use a channel and add a third case:

```
case <-c.remoteAlive:
   return
```

Reply:
With channel is cleaner anyways. Thanx.